### DISCUSSION TITLE
using only config.yaml to configure k3s

### DISCUSSION THREAD

**Discussion by cwrx777** [k3s-io/k3s#10345] [Q&A] [answered]
Hi 
is it possible to do the following CLI arguments and all others [server](https://docs.k3s.io/cli/server)/[agent ](https://docs.k3s.io/cli/agent) CLI arguments using only `config.yaml`  and how?
```
'--disable=servicelb' \
'--disable=traefik' \
'--data-dir=custom_dir' \
'--write-kubeconfig-mode=644' \
'--private-registry=custom_path_to_yaml>' \
'--embedded-registry' \
'--selinux' \
'--debug'
```

**Comment by brandond** [comment]
1. Yes
2. As documented here: https://docs.k3s.io/installation/configuration#configuration-file

**Reply by cwrx777** [reply]
Hi, 
the link is not clear on how to configure this kind of CLI argument:  `--disable=servicelb`, `--embedded-registry` in config file.
do we put it as 
```
servicelb: disabled
embedded-registry: true
```

It will be good if the key for the configuration file is also included in the[ CLI argument table](https://docs.k3s.io/cli/server#commonly-used-options) e.g.

| CLI Flag | Environment Variable |config file| 
|----------|-------------------------|---|
| `--write-kubeconfig-value, -o` value | `K3S_KUBECONFIG_OUTPUT` | `write-kubeconfig-value:` value | 
| `--embedded-registry` |  | `embedded-registry:` `true` | 
| `--disable=servicelb` |  | ??? |

**Reply by brandond** [reply]
It is covered in the docs page I linked.

> CLI arguments map to their respective YAML key, with repeatable CLI arguments being represented as YAML lists. Boolean flags are represented as true or false in the YAML file.

If that's not clear enough and you need an example:
* `--foo=bar` or `--foo bar` (these are equivalent forms) is `foo: bar`
* `--foo` (a boolean flag that takes no value) is `foo: true`
* `--foo=bar --foo=baz` is `foo: [bar, baz]` or any other valid yaml list expression

This is covered in similar examples on the docs page.

We are not going to list the config file key for each CLI arg, because they are **exactly** the same, it would be pointless duplication.

**Reply by cwrx777** [reply]
Hi,

then could you provide the config for

'--disable=servicelb' \
'--disable=traefik' \

**Reply by brandond** [reply, marked as answer]
I covered that already
> repeatable CLI arguments being represented as YAML lists.
> `--foo=bar --foo=baz` is `foo: [bar, baz]` or any other valid yaml list expression

So you could do
```yaml
disable: [servicelb, traefik]
```
or
```yaml
disable: 
  - servicelb
  - traefik
```

You might check out some yaml tutorials if you're not familiar with how lists are expressed in yaml. We don't expect to have to cover YAML basics in our documentation.
